### ISSUE TITLE
space after object class definition

### ISSUE THREAD

**Issue by aandyl** [cannatag/ldap3#344]
I have a server (OpenDJ) which is returning an object class definition with trailing space after the closing parenthesis, causing the recently-fixed schema validation to fail.

Do you know whether this whitespace is allowed by the RFC or not?

**Comment by cannatag** [comment]
Did it work with the previous version of ldap3? Are you sure that the problem is in the trailing space? The latest fixes on validation schema were meant to extend compatibility with non Unicode descriptions in the schema, the parser has not changed.


Can you give me information on how to replicate the problem in my lab? I need the python version you're using, the openDJ version and a snippet of code to show the problem. Also, can you tell me if the problem is caused by a standard object class or by a custom one?

Thanks,
Giovanni

**Comment by cannatag** [comment]
Please, disregard my previous message. I've installed a docker image of OpenDJ 2.6 and have found the "offending" objectClass definition in the schema. RFC 4512 states that the right-parenthesis should be the last character in the definition:
```
     ObjectClassDescription = LPAREN WSP
         numericoid                 ; object identifier
         [ SP "NAME" SP qdescrs ]   ; short names (descriptors)
         [ SP "DESC" SP qdstring ]  ; description
         [ SP "OBSOLETE" ]          ; not active
         [ SP "SUP" SP oids ]       ; superior object classes
         [ SP kind ]                ; kind of class
         [ SP "MUST" SP oids ]      ; attribute types
         [ SP "MAY" SP oids ]       ; attribute types
         extensions WSP RPAREN
```
so that definition should not be valid, anyway I've released the ldap3 parser to strip any leading and trailing spaces in each object definition in the schema. Now it should work, can you check the code in dev?

Thanks,
Giovanni

**Comment by andifex** [comment]
I checked with the latest dev branch and it is indeed working now with my server, thanks for the fast response!

**Comment by aandyl** [comment]
I checked with the latest dev branch and it is indeed working now with my server, thanks for the fast response!

**Comment by cannatag** [comment]
Good. The fiducia silo be release in the ne trovò version of ldap3.

**Comment by cannatag** [comment]
Fix released in ldap3 version 2.2.4
